### PR TITLE
fix mcp.types.Tool is not following python's naming convention snake_case 

### PR DESCRIPTION
**Pull request by azdolinski**
Renamed the `inputSchema` field to `input_schema` in the `Tool` class to align with Python's naming conventions (snake_case). Added an alias `inputSchema` for backward compatibility.

## Motivation and Context

The name `inputSchema` does not follow Python's naming convention (PEP 8). This change improves code readability and consistency. Fix related: #97 

## How Has This Been Tested?

Tested locally by running existing tests.

## Breaking Changes

- [X] Yes. Users will need to update references to the `inputSchema` field to `input_schema`. Adding an alias is intended to ease the migration.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update

## Checklist

- [X] I have read the [MCP Documentation](https://modelprotocol.io)
- [X] My code follows the repository's style guidelines
- [X] New and existing tests pass locally
- [X] I have added appropriate error handling
- [X] I have added or updated documentation as needed

## Additional context

The addition of the `inputSchema` alias is intended to provide backward compatibility and facilitate the migration process for users.

### CODE REVIEW

**Comment by dsp-ant** [comment]
Hmm I think this is sensible, but we need to probably wait for SDK 2.0 to introduce this as it is a BC break. I keep it open for now until we start a 2.0 branch.

**Comment by azdolinski** [comment]
@dsp-ant 

OK.. Fully understood.
Just for considered: as alternative which could be not BC.

```python
class Tool(BaseModel):
    """Definition for a tool the client can call."""

    name: str
    """The name of the tool."""
    description: str | None = None
    """A human-readable description of the tool."""
    input_schema: dict[str, Any]
    """A JSON Schema object defining the expected parameters for the tool."""
    model_config = ConfigDict(extra="allow", populate_by_name=True)

    @model_validator(mode='before')
    @classmethod
    def _convert_camel_to_snake_names(cls, data):
        if isinstance(data, dict) and 'inputSchema' in data:
            data['input_schema'] = data.pop('inputSchema')
        return data
```


The main differences between the original `Tool` class and this one

1. **Field Naming Convention**:
   - In the original version, the field is named `inputSchema` (camelCase).
   - In the this version, the field is renamed to `input_schema` (snake_case), which is more Pythonic.

2. **Populate by Name**:
   - This version includes `populate_by_name=True` in the `model_config`. This allows the model to accept both `inputSchema` and `input_schema` as valid field names during initialization.

3. **Model Validator**:
   - In this version we adds a `@model_validator` method (`_convert_camel_to_snake_names`) to automatically convert `inputSchema` to `input_schema` if the input data uses the camelCase version. This ensures backward compatibility with data using the old naming convention.

### Example:
```python
# Original version work as:
tool1 = Tool(name="hammer", inputSchema={"type": "object"})

# Proposed version should work in both cases
tool2 = Tool(name="hammer", inputSchema={"type": "object"})  # Still works
tool3 = Tool(name="hammer", input_schema={"type": "object"})  # Also works
```

In summary, the second version is more flexible and handles both naming conventions seamlessly. 
Could be applied also in case of `mimeType`

**Comment by dsp-ant** [comment]
> @dsp-ant
> 
> OK.. Fully understood. Just for considered: as alternative which could be not BC.
> 
> ```python
> class Tool(BaseModel):
>     """Definition for a tool the client can call."""
> 
>     name: str
>     """The name of the tool."""
>     description: str | None = None
>     """A human-readable description of the tool."""
>     input_schema: dict[str, Any]
>     """A JSON Schema object defining the expected parameters for the tool."""
>     model_config = ConfigDict(extra="allow", populate_by_name=True)
> 
>     @model_validator(mode='before')
>     @classmethod
>     def _convert_camel_to_snake_names(cls, data):
>         if isinstance(data, dict) and 'inputSchema' in data:
>             data['input_schema'] = data.pop('inputSchema')
>         return data
> ```
> 
> The main differences between the original `Tool` class and this one
> 
> 1. **Field Naming Convention**:
>    
>    * In the original version, the field is named `inputSchema` (camelCase).
>    * In the this version, the field is renamed to `input_schema` (snake_case), which is more Pythonic.
> 2. **Populate by Name**:
>    
>    * This version includes `populate_by_name=True` in the `model_config`. This allows the model to accept both `inputSchema` and `input_schema` as valid field names during initialization.
> 3. **Model Validator**:
>    
>    * In this version we adds a `@model_validator` method (`_convert_camel_to_snake_names`) to automatically convert `inputSchema` to `input_schema` if the input data uses the camelCase version. This ensures backward compatibility with data using the old naming convention.
> 
> ### Example:
> ```python
> # Original version work as:
> tool1 = Tool(name="hammer", inputSchema={"type": "object"})
> 
> # Proposed version should work in both cases
> tool2 = Tool(name="hammer", inputSchema={"type": "object"})  # Still works
> tool3 = Tool(name="hammer", input_schema={"type": "object"})  # Also works
> ```
> 
> In summary, the second version is more flexible and handles both naming conventions seamlessly. Could be applied also in case of `mimeType`

sorry totally missed this. I think we should probably do this in the v1.x series and start warning with a deprecation warning if inputSchema is used. Do you mind sending a PR and re-requesting a review @azdolinski ?

**Comment by felixweinberger** [comment]
Thanks for this @azdolinski we'll want to do this for the whole code base and given it's likely to be a breaking change we're punting this to v2 of the SDK.

**Comment by felixweinberger** [comment]
Closing this as we're tracking in https://github.com/modelcontextprotocol/python-sdk/issues/1253
